bundle/DependencyInjection/Configuration.php

```diff
@@ -73,4 +75,94 @@ private function addAsynchronousIndexingSection(ArrayNodeDefinition $nodeDefinit
                 ->end()
             ->end();
     }
+
+    private function addPageIndexingSection(ArrayNodeDefinition $nodeDefinition): void
+    {
+        $keyValidator = static function ($v) {
+            foreach (array_keys($v) as $key) {
+                if (!is_string($key)) {
+                    return true;
+                }
+            }
+            return false;
+        };
+        $nodeDefinition
+            ->children()
+                ->arrayNode('page_indexing')
+                ->addDefaultsIfNotSet()
+                ->info('Page indexing configuration')
+                ->children()
+                    ->booleanNode('enabled')
+                        ->info('Use layouts page text indexing')
+                        ->defaultFalse()
+                    ->end()
+                    ->arrayNode('sites')
+                        ->useAttributeAsKey('name')
+                        ->normalizeKeys(false)
+                        ->validate()
+                            ->ifTrue($keyValidator)
+                            ->thenInvalid('Site root name must be of string type.')
+                        ->end()
+                        ->arrayPrototype()
+                            ->children()
+                                ->integerNode('tree_root_location_id')
+                                    ->info('Site root id')
+                                    ->beforeNormalization()->always(static fn ($v) => is_string($v) ? (int)$v : $v)->end()
+                                ->end()
+                                ->arrayNode('languages_siteaccess_map')
+                                    ->info('Language key mapped to page siteaccess')
+                                    ->useAttributeAsKey('name')
+                                    ->normalizeKeys(false)
+                                    ->validate()
+                                        ->ifTrue($keyValidator)
+                                        ->thenInvalid('Page name must be of string type.')
```

Comment:
"Language code must be of string type".

Reply:
fixed in 9e2818fbb70808cd454e96b7ac239d701d99bd7a